src/test/HST/ApplicationTests.hs

```diff
@@ -113,6 +137,40 @@ testProcessModule = context "processModule" $ do
     , "      a4 = undefined"
     , "  in  a3"
     ]
+  it "should transform a module which uses a data type from a different module"
+    $ runTest
+    $ let ms = [ [ "module B where"
+                   , "import A"
+                   , "mapTree f (Leaf x) = Leaf (f x)"
+                   , "mapTree f (Branch l r) = Branch (mapTree f l) (mapTree f r)"
+                   ]
+               , [ "module A where"
+                   , "data Tree a = Leaf a | Branch (Tree a) (Tree a)"
+                   ]
+               ]
+          m  = [ "module B where"
+               , "import A"
+               , "mapTree a0 a1 = case a1 of"
+               , "  Leaf a2      -> Leaf (a0 a2)"
+               , "  Branch a4 a5 -> Branch (mapTree a0 a4) (mapTree a0 a5)"
+               ]
+      in shouldTransformModulesTo ms m
+  it "should not change the name of a function defined in another module"
+    $ runTest
+    $ let ms = [ [ "module B where"
+                   , "import A"
+                   , "incList [] = []"
+                   , "incList (n:ns) = inc n : incList ns"
+                   ]
+               , ["module A where", "inc n = n + 1"]
+               ]
+          m  = [ "module B where"
+               , "import A"
+               , "incList a0 = case a0 of"
+               , "  [] -> []"
+               , "  a1:a2 -> inc a1 : incList a2"
+               ]
+      in shouldTransformModulesTo ms m
```

Comment:
After the branch for qualified imports has been merged into this one, we should add at least one test that checks if we parse qualified imports, aliased imports and multiple imports for the same module correctly. I can add this or these if you want.